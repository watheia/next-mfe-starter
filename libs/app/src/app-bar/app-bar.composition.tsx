import { AuthProvider, SupabaseProvider } from '@watheia/mfe.api';
import React from 'react';
import AppBar from './app-bar';

export const AppBarExample = () => {
  return (
    <SupabaseProvider
      publicUrl={process.env['NEXT_PUBLIC_SUPABASE_URL'] ?? ''}
      anonKey={process.env['NEXT_PUBLIC_SUPABASE_ANON_KEY'] ?? ''}
    >
      <AuthProvider>
        <AppBar />
      </AuthProvider>
    </SupabaseProvider>
  );
};
