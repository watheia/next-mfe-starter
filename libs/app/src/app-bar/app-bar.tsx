import MuiAppBar from '@mui/material/AppBar';
import Button from '@mui/material/Button';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import { useAuth, url } from '@watheia/mfe.api';
import { fixtures } from '@watheia/mfe.model';
import { Link } from '@watheia/ui-atoms';
import { TabNav } from '@watheia/ui-organisms';
import { useRouter } from 'next/router';
import React, { MouseEvent } from 'react';

/* eslint-disable-next-line */
export interface AppBarProps {}

export function AppBar(props: AppBarProps) {
  const { session, signOut } = useAuth();
  const router = useRouter();

  const signoutHandler = async (e: MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    signOut().then(() => router.push('/'));
  };

  return (
    <MuiAppBar
      position="static"
      color="transparent"
      elevation={0}
      sx={{
        borderBottom: (theme) => `1px solid ${theme.palette.divider}`,
        backgroundColor: (theme) => theme.palette.background.default,
      }}
    >
      <Toolbar sx={{ flexWrap: 'wrap' }}>
        <Typography variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
          <Link variant="inherit" color="text.primary" href={url().href} underline="hover">
            waweb
          </Link>
        </Typography>
        {session && (
          <>
            <TabNav items={fixtures.primaryNav} />
            <Button
              onClick={signoutHandler}
              variant="text"
              color="inherit"
              sx={{ my: 1, mx: 1.5 }}
            >
              Logout
            </Button>{' '}
          </>
        )}
      </Toolbar>
    </MuiAppBar>
  );
}

export default AppBar;
