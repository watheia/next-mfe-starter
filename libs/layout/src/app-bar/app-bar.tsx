import MuiAppBar from '@mui/material/AppBar';
import Button from '@mui/material/Button';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import { Link } from '@waweb/atoms';
import { TabNav } from '@waweb/organisms';
import { getUrl } from '@waweb/utils';
import { fixtures } from '@waweb/model';
import React from 'react';

/* eslint-disable-next-line */
export interface AppBarProps {}

export function AppBar(props: AppBarProps) {
  return (
    <MuiAppBar
      position="static"
      color="primary"
      elevation={0}
      sx={{
        borderBottom: (theme) => `1px solid ${theme.palette.divider}`,
        backgroundColor: (theme) => theme.palette.background.default,
      }}
    >
      <Toolbar sx={{ flexWrap: 'wrap' }}>
        <Typography variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
          <Link variant="inherit" color="text.primary" href={getUrl()} underline="hover">
            waweb
          </Link>
        </Typography>
        <TabNav routes={fixtures.primaryNav} />
        <Button
          href={getUrl('/auth')}
          color="primary"
          variant="text"
          sx={{ my: 1, mx: 1.5 }}
        >
          Login
        </Button>
      </Toolbar>
    </MuiAppBar>
  );
}

export default AppBar;
