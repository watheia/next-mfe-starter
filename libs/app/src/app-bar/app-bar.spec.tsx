import { render } from '@testing-library/react';

import { AppBarExample } from './app-bar.composition';

describe('AppBar', () => {
  it('should render successfully', () => {
    const { container } = render(<AppBarExample />);
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
});
