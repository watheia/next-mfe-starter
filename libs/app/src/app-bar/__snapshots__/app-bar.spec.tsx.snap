// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render successfully 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-jyud63-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t018ql-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1a3i5r6-MuiTypography-root-MuiLink-root"
          href="http://localhost:4200/"
        >
          waweb
        </a>
      </h6>
    </div>
  </header>
</div>
`;
